tests/js/tab/tab-button-tests.js

```diff
@@ -0,0 +1,226 @@
+            setTimeout(function () {
+                // Restore the gamepad back to its neutral state.
+                gamepad.tests.navigator.pollGamepads();
+
+                // Check if the last element is focused.
+                jqUnit.assertEquals("The last element should be focused.", document.querySelector("#last"), document.activeElement);
+                jqUnit.start();
+            }, gamepad.tests.frequency * 3);
+        });
+
+        jqUnit.asyncTest("Change the focus to one of the next elements in forward tabbing using buttons.", function () {
+            gamepad.tests.windowObject.navigator.getGamepads = function () {
+                return gamepad.tests.utils.buttons.tab(5);
+            };
+
+            // Set initial conditions i.e., focus on the first element.
+            $("#first").focus();
+
+            // Confirm that the instance of the gamepad navigator is created.
+            gamepad.tests.navigator = gamepad.inputMapper({
+                windowObject: gamepad.tests.windowObject,
+                frequency: gamepad.tests.frequency
+            });
+            jqUnit.assertTrue("The Gamepad Navigator should be instantiated.", fluid.isComponent(gamepad.tests.navigator));
+
+            // Check the state of gamepad inputs and webpage after polling.
+            gamepad.tests.navigator.pollGamepads();
+            jqUnit.assertLeftHand("The gamepad should be connected with no buttons/axes disturbed initially.", gamepad.tests.modelAtRest, gamepad.tests.navigator.model);
+            jqUnit.assertEquals("The focus should not be changed after polling.", document.querySelector("#first"), document.activeElement);
+
+            // Record the state of focused elements before polling.
+            var beforePollingFocusedElementTabIndex = document.activeElement.getAttribute("tabindex");
+
+            /**
+             * Update the gamepad to press button 5 (right bumper) for forward tab
+             * navigation.
+             */
+            gamepad.tests.navigator.pollGamepads();
+
+            // Wait for a few milliseconds for the navigator to change focus.
+            setTimeout(function () {
+                // Restore the gamepad back to its neutral state.
+                gamepad.tests.navigator.pollGamepads();
+
+                // Record the index of the element currently focused.
+                var afterPollingFocusedElementTabIndex = document.activeElement.getAttribute("tabindex");
+
+                // Check if the focus has moved in the forward direction.
+                var hasTabbedForward = beforePollingFocusedElementTabIndex < afterPollingFocusedElementTabIndex;
+                jqUnit.assertTrue("The focus should have moved in the forward direction.", hasTabbedForward);
+                jqUnit.start();
+            }, gamepad.tests.frequency * 4);
+        });
+
+        jqUnit.asyncTest("Change the focus to one of the previous elements in reverse tabbing using buttons.", function () {
```

Comment:
I think it would be better to be more specific than "button".  I would suggest using "continuous" and "discrete".  "Continuous" would be for held buttons where the action continues.  "Discrete" would be for activities that happen once when a button is pressed and released.  Once we get to the point where you have a control panel, you'll want to let people choose which style of navigation to use.  For example, people with limited motor control may need "discrete" controls to avoid triggering an action twice before they can release the button.